### PR TITLE
MB-62230: Added params in KNN request when parsing a query

### PR DESCRIPTION
**Pull request by metonymic-smokey**
This PR adds support for search-time params in the query parser and custom JSON unmarshaling logic.

### CODE REVIEW

File: search_knn.go [hunk 1]
```diff
@@ -387,7 +389,7 @@ func (i *indexImpl) runKnnCollector(ctx context.Context, req *SearchRequest, rea
 		}
 
 		// Applies to all supported types of queries.
-		filterSearcher, _ := filterQ.Searcher(ctx, reader, i.m, search.SearcherOptions{
+		filterSearcher, _ := filterQ.(query.Query).Searcher(ctx, reader, i.m, search.SearcherOptions{
```
Comment (abhinavdangeti) [inline search_knn.go:392]: If this type assertion were to fail, wouldn't we be dereferencing a `nil` pointer.

File: search_knn.go [hunk 2]
```diff
@@ -196,6 +197,7 @@ func (r *SearchRequest) UnmarshalJSON(input []byte) error {
 		r.KNN[i].VectorBase64 = temp.KNN[i].VectorBase64
 		r.KNN[i].K = temp.KNN[i].K
 		r.KNN[i].Boost = temp.KNN[i].Boost
+		r.KNN[i].Params = temp.KNN[i].Params
```
Comment (abhinavdangeti) [inline search_knn.go:200]: Was this a regression from the previous change?

**Review by abhinavdangeti — CHANGES_REQUESTED** [review]

File: search_knn.go (hunk 1 above)
Comment (metonymic-smokey) [reply search_knn.go:392]: yes, was _just_ pushing a commit up for this - gimme a min

File: search_knn.go (hunk 2 above)
Comment (metonymic-smokey) [reply search_knn.go:200]: yes, fixed in a recent commit.

**Review by abhinavdangeti — APPROVED** [review]

**Pull request merged**